rosplan_knowledge_base/src/VALVisitorOperator.cpp

```diff
@@ -128,6 +128,7 @@ namespace KCL_rosplan {
 
 	void VALVisitorOperator::visit_effect_lists(VAL::effect_lists * e) {
 
+		eff_time = VAL::E_AT_END;  // Set non-timed add and del effects to be at end in the ROSPlan msgs
```

Comment:
I think this may still not be 100% fixed, for the following reason. Actions are split into two nodes (start and end) and the conditions and effects are assigned, then used to derive a partial order between nodes. If an instantaneous action becomes two nodes, (a+b) and another action (c) interferes with its effects, then they should be ordered. Suppose c appears before (a+b) in the plan, we want to see the ordering on nodes c<<a. However, since the ordering is performed on the node level, we might see the ordering c<<b. That is, (a) can trigger before (c).

I need to look at the plan parser to see if the node (c) will be ordered before the start node if a conflict is found between the end, but I don't think so (as it wouldn't be correct).

Reply:
It is exactly the same bug as putting the effects in the start, but I think far less common. Actions will dispatch in parallel when they are not supposed to. As this is at the least and improvement, and maybe all we need, I would prefer to open a new issue to check this and merge the PR anyway.